File: src/main/java/link/rdcn/user/Credentials.java
package link.rdcn.user;

import java.io.Serializable;
import java.util.Objects;

/**
 * @Author renhao
 * @Description:
 * @Data 2025/6/24 11:00
 * @Modified By:
 */
public class Credentials implements Serializable {

    private final boolean anonymous;

    public static final Credentials ANONYMOUS = new Credentials(true);

    public Credentials() {
        this(false);
    }

    private Credentials(boolean anonymous) {
        this.anonymous = anonymous;
    }

    public boolean isAnonymous() {
        return anonymous;
    }
}

File: src/main/scala/link/rdcn/user/Credentials.scala
package link.rdcn.user

/**
 * @Author renhao
 * @Description:
 * @Data 2025/7/9 16:55
 * @Modified By:
 */
trait Credentials extends Serializable

object Credentials {
  case object ANONYMOUS extends Credentials

  def anonymous: Credentials = ANONYMOUS
}

